### PR TITLE
Handle eth bribes from hidden hand

### PR DESCRIPTION
**Pull request by shuklaayush**
- Fix bribe processing to handle ETH
- Add tests via mock contracts

### CODE REVIEW

File: contracts/MyStrategy.sol [hunk 1]
```diff
@@ -400,4 +426,11 @@ contract MyStrategy is BaseStrategy, ReentrancyGuardUpgradeable {
         IERC20Upgradeable(BADGER).safeTransfer(BADGER_TREE, amount);
         _processExtraToken(address(BADGER), amount);
     }
-}
+
+    /// PAYABLE FUNCTIONS ///
+
+    /// @dev Can only receive ether from Hidden Hand
+    receive() external payable {
```
Comment (GalloDaSballo) [inline contracts/MyStrategy.sol:433]: Can we have a storage flag to ensure this is called also only by harvest?

Something like `claiming = true`
`receive -> require(claiming)`

**Review by GalloDaSballo — COMMENTED** [review]
Beside the note on receive LG

File: contracts/MyStrategy.sol (hunk 1 above)
Comment (GalloDaSballo) [reply contracts/MyStrategy.sol:433]: wdyt?

Comment (shuklaayush) [reply contracts/MyStrategy.sol:433]: I guess that makes more sense. I'll do that instead

Comment (shuklaayush) [reply contracts/MyStrategy.sol:423]: How about now? I also removed the eth sweeping function since that probably won't be required now

**Comment by sajanrajdev** [comment]
@shuklaayush Not a part of this PR but it seems like the parameter and second return value here are not used:
https://github.com/Badger-Finance/vested-aura/blob/6628f6f31a4a39d66beb0bbefc47ca55d9e3623d/contracts/MyStrategy.sol#L388-L391

**Review by sajanrajdev — APPROVED** [review]
Introduces handling of ETH bribes receiving from HH. Contract is only payable while claiming rewards, so only from HH. ETH is wrapped and handled like the rest of the rewards through the processor.

Tests were included for these functionalities.

LGTM
image

**Pull request merged**